OA_Core.Repository/Repositories/AulaRepository.cs

```diff
@@ -0,0 +1,98 @@
+﻿using Microsoft.EntityFrameworkCore;
+using OA_Core.Domain.Entities;
+using OA_Core.Domain.Interfaces.Repository;
+using OA_Core.Repository.Context;
+using MySqlConnector;
+namespace OA_Core.Repository.Repositories
+{
+    public class AulaRepository : IAulaRepository
+    {
+        private readonly CoreDbContext _context;
+
+        public AulaRepository(CoreDbContext context)
+        {
+            _context = context;
+        }
+
+        public async Task AddAsync(Aula aula)
+        {
+            var sql = "INSERT INTO Aula VALUES(@id, @nome, @descricao, @caminho, @tipo, @duracao, @ordem, @curso_id, @data_criacao, @data_alteracao, @data_delecao)";
+            object[] paramItems = new object[]
+            {
+                new MySqlParameter("@id", aula.Id),
+                new MySqlParameter("@nome", aula.Nome),
+                new MySqlParameter("@descricao", aula.Descricao),
+                new MySqlParameter("@caminho", aula.Caminho),
+                new MySqlParameter("@tipo", aula.Tipo),
+                new MySqlParameter("@duracao", aula.Duracao),
+                new MySqlParameter("@ordem", aula.Ordem),
+                new MySqlParameter("@curso_id", aula.CursoId),
+                new MySqlParameter("@data_criacao", aula.DataCriacao),
+                new MySqlParameter("@data_alteracao", aula.DataAlteracao),
+                new MySqlParameter("@data_delecao", aula.DataDelecao)
+            };
+            await _context.Database.ExecuteSqlRawAsync(sql, paramItems);
+        }
+
+        public async Task EditAsync(Aula aula)
+        {
+            var sql = "UPDATE Aula SET id = @id, nome = @nome, descricao = @descricao, caminho = @caminho, tipo = @tipo, duracao = @duracao, ordem = @ordem, curso_id = @curso_id, data_criacao = @data_criacao, data_alteracao = @data_alteracao, data_delecao = @data_delecao WHERE id = @id";
+            object[] paramItems = new object[]
+            {
+                new MySqlParameter("@id", aula.Id),
+                new MySqlParameter("@nome", aula.Nome),
+                new MySqlParameter("@descricao", aula.Descricao),
+                new MySqlParameter("@caminho", aula.Caminho),
+                new MySqlParameter("@tipo", aula.Tipo),
+                new MySqlParameter("@duracao", aula.Duracao),
+                new MySqlParameter("@ordem", aula.Ordem),
+                new MySqlParameter("@curso_id", aula.CursoId),
+                new MySqlParameter("@data_criacao", aula.DataCriacao),
+                new MySqlParameter("@data_alteracao", aula.DataAlteracao),
+                new MySqlParameter("@data_delecao", aula.DataDelecao)
+            };
+            await _context.Database.ExecuteSqlRawAsync(sql, paramItems);
+        }
+
+        public async Task<Aula> FindAsync(Guid id)
+        {
+            var query = "SELECT * FROM Aula WHERE id = @id AND data_delecao is null";
```

Comment:
Eu não acho uma boa prática utilizar o "*", porque como o banco não está no código, torna menos legível não especificar os campos. Além disso, mais para frente o ideal é especializar esses repositórios para trazer só os campos uteis conforme o resultado esperado.

Reply:
No momento, as consultas por meio de query estão exigindo, devido algum mapeamento, que as mesmas sejam feitas, buscando por todos os campos da entidade, onde, caso seja feita uma consulta sem o data_alteracao, data_criacao e demais campos, aconteça o seguinte erro:
MySqlConnector.MySqlException (0x80004005): Unknown column 'o.data_alteracao' in 'field list'
e outros erros semelhantes, por conta disso, acaba sendo mais viável utilizar do '*', já que todos os campos da entidade irão ser buscados, e após isso, ignorados devido a conversão de contratos dessas entidades pela camada de serviço.